pootle/apps/import_export/utils.py

```diff
@@ -49,9 +50,16 @@ def import_file(file, user=None):
                                 "Project/Language? (%s)", (file.name, e)))
 
     try:
+        tp = store.translation_project
+        allow_add_and_obsolete = ((tp.project.checkstyle == 'terminology'
+                                   or tp.is_template_project)
+                                  and check_user_permission(user,
+                                                            'administrate',
+                                                            tp.directory))
```

Comment:
i think these lines should be outside the try...except - as that is there to catch Store.update errors